### PR TITLE
Download graph image

### LINKED ISSUE #109: Download dependency graph
There should be a button allowing to download the current view of the dependency graph as image file.

### PR DESCRIPTION
**Pull request by georg-schwarz**
Closes #109.
Builds on #108. Only merge after!

Added a download button that downloads the current view of the dependency graph.
Better diff: https://github.com/osrgroup/msadoc/compare/105-configure-graph-depth...109-download-graph-image

## Screenshot
image

### CODE REVIEW

File: frontend/src/components/main-page/graph-page/dependency-graph-view.tsx [hunk 1]
```diff
@@ -53,41 +59,112 @@ export const DependencyGraph: React.FC = () => {
...
+    const pngBlob = await cyInstance.png({
+      output: 'blob-promise',
+    });
+
+    const fileName = 'msadoc-dependency-graph.png';
+    const downloadLink = document.createElement('a');
+    downloadLink.href = URL.createObjectURL(pngBlob);
+    downloadLink.download = fileName;
+    downloadLink.click();
+  };
```
Comment (MBuchalik) [inline frontend/src/components/main-page/graph-page/dependency-graph-view.tsx:77]: I would move this (and the entire rest of the logic) into the Controller. And maybe call this function "performDownload" or "initiateDownload" or "downloadAsPNG" or sth like that.

**Review by MBuchalik — COMMENTED** [review]
Nice 👍

File: frontend/src/components/main-page/graph-page/dependency-graph-view.tsx (hunk 1 above)
Comment (georg-schwarz) [reply frontend/src/components/main-page/graph-page/dependency-graph-view.tsx:77]: The issue is that I need the reference to the cytoscape instance.. Should I put that into the controller/state as well? We get the reference by a callback:

```html
<CytoscapeComponent
    ...
    cy={(cy): void => {
      // save the reference cy in the state?
    }}
/>
```

Comment (georg-schwarz) [reply frontend/src/components/main-page/graph-page/dependency-graph-view.tsx:78]: Tried it saving in the state: endless loop.

Rendered `CytoscapeComponent` sets the state, leading to rendering `CytoscapeComponent` and so forth :)

Comment (georg-schwarz) [reply frontend/src/components/main-page/graph-page/dependency-graph-view.tsx]: Awesome, thx!

**Pull request merged**